src/core/stream_recv.c

```diff
@@ -162,6 +162,83 @@ QuicStreamRecvQueueFlush(
     }
 }
 
+//
+// Deliver a notification to the app that the peer has aborted their send path.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicStreamAlertAppPeerSendAborted(
+    _In_ QUIC_STREAM* Stream,
+    _In_ QUIC_VAR_INT ErrorCode
+    )
+{
+    QuicTraceLogStreamInfo(
+        RemoteCloseReset,
+        Stream,
+        "Closed remotely (reset)");
+    QUIC_STREAM_EVENT Event;
+    Event.Type = QUIC_STREAM_EVENT_PEER_SEND_ABORTED;
+    Event.PEER_SEND_ABORTED.ErrorCode = ErrorCode;
+    QuicTraceLogStreamVerbose(
+        IndicatePeerSendAbort,
+        Stream,
+        "Indicating QUIC_STREAM_EVENT_PEER_SEND_ABORTED (0x%llX)",
+        ErrorCode);
+    (void)QuicStreamIndicateEvent(Stream, &Event);
+}
+
+//
+// Processes a received RELIABLE_RESET frame's payload.
+//
+_IRQL_requires_max_(PASSIVE_LEVEL)
+void
+QuicStreamProcessReliableResetFrame(
+    _In_ QUIC_STREAM* Stream,
+    _In_ QUIC_VAR_INT ErrorCode,
+    _In_ QUIC_VAR_INT ReliableOffset
+    )
+{
+    if (!Stream->Connection->State.ReliableResetStreamNegotiated) {
+        //
+        // The peer tried to use an exprimental feature without
+        // negotiating first. Kill the connection.
+        //
+        QuicTraceLogStreamWarning(
+            ReliableResetNotNegotiatedError,
+            Stream,
+            "Received ReliableReset without negotiation.");
+        QuicConnTransportError(Stream->Connection, QUIC_ERROR_TRANSPORT_PARAMETER_ERROR);
+        return;
+    }
+
+    if (Stream->RecvMaxLength == 0 || ReliableOffset < Stream->RecvMaxLength) {
+        //
+        // As outlined in the spec, if we receive multiple CLOSE_STREAM frames, we only accept strictly
+        // decreasing offsets.
+        //
+        Stream->RecvMaxLength = ReliableOffset;
+        Stream->Flags.RemoteCloseResetReliable = TRUE;
+
+        QuicTraceLogStreamInfo(
+            ReliableRecvOffsetSet,
+            Stream,
+            "Reliable recv offset set to %llu",
+            ReliableOffset);
+    }
+
+    if (Stream->RecvBuffer.BaseOffset >= Stream->RecvMaxLength) {
+        QuicTraceEvent(
+            StreamRecvState,
+            "[strm][%p] Recv State: %hhu",
+            Stream,
+            QuicStreamRecvGetState(Stream));
+        QuicStreamAlertAppPeerSendAborted(Stream, ErrorCode);
+        QuicStreamRecvShutdown(Stream, TRUE, ErrorCode);
+    } else {
+        Stream->RecvShutdownErrorCode = ErrorCode;
```

Comment:
This is worth a comment along the lines of "we still have data to deliver to the app, just cache the error code for later".

Reply:
Added